pgjdbc/src/test/java/org/postgresql/core/ParserTest.java

```diff
@@ -156,6 +156,51 @@ public void testModifyJdbcCall() throws SQLException {
     assertEquals("call lower(?,?)", Parser.modifyJdbcCall("{call lower(?,?)}", true, ServerVersion.v11.getVersionNum(), 3, EscapeSyntaxCallMode.CALL).getSql());
   }
 
+  /**
+   * Tests whether a call statement to a function is recognised as function call.
+   * @throws SQLException if parsing fails due to SQL error
+   */
+  @Test
+  public void testModifyJdbcCallRecogniseFunctionCall() throws SQLException {
+    assertIsFunction(Parser.modifyJdbcCall("{ ? = call test_function(?)}", true, ServerVersion.v14.getVersionNum(), 3, EscapeSyntaxCallMode.CALL));
```

Comment:
do you think you could push `Parser.modifyJdbcCall` into `assertIsFunction` as well, so its failure message would indicate the way it called parsing?

Reply:
I moved the call to `assertIsFunction`, which is a good idea since it reduces duplicate code. However, I don't see how this would improve the failure message.
Do you mind if I switch this test class `ParserTest` to JUnit5 instead of JUnit4?
Then we could use a `@ParemeterizedTest` annotated test method here instead of 4 test methods. This would also allow to add more parameters easily, e.g. the server version, though I doubt that this would impact this issue.

Reply:
> which is a good idea since it reduces duplicate code

Glad you liked it. Every time I ask I am afraid it would be considered splitting hairs.

>However, I don't see how this would improve the failure message.

How about the following?

```java
    JdbcCallParseInfo jdbcCallParseInfo = Parser.modifyJdbcCall(sql, true, ServerVersion.v14.getVersionNum(), 3, EscapeSyntaxCallMode.CALL);
    String message = "Parser.modifyJdbcCall(\"" + sql + "\", , true, ServerVersion.v14.getVersionNum(), 3, EscapeSyntaxCallMode.CALL).isFunction was supposed to return FUNCTION, however got " + jdbcCallParseInfo;
```

----

Switch to JUnit5 is just fine.
Frankly speaking, I would love to rewrite the existing `BaseTest4` into something like JUnit5 extension, however, I had no chances to do that yet.

Reply:
`JdbcCallParseInfo` does not implement `toString()` obviously:
```
Parser.modifyJdbcCall("/* some comment */ { ? = call test_function(?)}", , true, ServerVersion.v14.getVersionNum(), 3, EscapeSyntaxCallMode.CALL).isFunction was supposed to return FUNCTION, however got org.postgresql.core.JdbcCallParseInfo@5f1b1944 expected:<true> but was:<false>
```

Therefore I omitted the _however ..._ part:
```
Parser.modifyJdbcCall("/* some comment */ { ? = call test_function(?)}", , true, ServerVersion.v14.getVersionNum(), 3, EscapeSyntaxCallMode.CALL).isFunction was supposed to return FUNCTION,  expected:<true> but was:<false>
```

Reply:
> Switch to JUnit5 is just fine. Frankly speaking, I would love to rewrite the existing `BaseTest4` into something like JUnit5 extension, however, I had no chances to do that yet.

I have _converted_ the `ParserTest` to JUnit5 (update imports, swap method params) and changed the four test methods to one ParameterizedTest - so far only locally, because the `ParameterizedTest` is not executed and I have no idea why.

Reply:
> Switch to JUnit5 is just fine. Frankly speaking, I would love to rewrite the existing `BaseTest4` into something like JUnit5 extension, however, I had no chances to do that yet.

I have _converted_ the `ParserTest` to JUnit5 (update imports, swap method params) and changed the four test methods to one ParameterizedTest.